### PR TITLE
bpo-46290: Fix parameter names in dataclasses docs

### PR DESCRIPTION
**Pull request by zsol**
Fix documentation typos in parameter names for dataclasses API.
These are important to avoid confusion when passing in arguments as keywords, like in https://github.com/Instagram/LibCST/pull/585#discussion_r780154212

https://bugs.python.org/issue46290

### CODE REVIEW

File: Misc/NEWS.d/next/Documentation/2022-01-07-10-23-56.bpo-46290.UKZFmh.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix parameter names in the dataclasses module's documentation.
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Documentation/2022-01-07-10-23-56.bpo-46290.UKZFmh.rst]:
```suggestion
Fix parameter names in the :mod:`dataclasses` documentation.
```

File: Doc/library/dataclasses.rst [hunk 2]
```diff
@@ -406,10 +406,10 @@ Module contents
          def add_one(self):
              return self.x + 1
 
-.. function:: replace(instance, /, **changes)
+.. function:: replace(obj, /, **changes)
```
Comment (merwok) [inline Doc/library/dataclasses.rst:409]: FTR the slash marks the arguments before it as positional-only, so the name chosen does not matter as it does not apply to code, it’s only for readers of the doc.  (But if all other instances are changed, it makes sense to change this one too.)

Comment (zsol) [reply Doc/library/dataclasses.rst:409]: Yeah this is more for consistency than anything else.

Comment (taleinat) [reply Doc/library/dataclasses.rst:409]: The first argument for the other functions, `asdict()`, `astuple()`, and `isdataclass()`, is *not* positional-only. So changing the argument name is indeed a meaningful fix that should be made. In that context, indeed changing the name for `replace()` for consistency makes sense.

**Review by taleinat — APPROVED** [review]

**Comment by taleinat** [comment]
Something the NEWS check doesn't know is that we don't add NEWS entries for documentation-only fixes. I'm adding the appropriate label to this PR and removing the NEWS entry.

**Comment by zsol** [comment]
Thanks! I saw somewhere that minor typo changes generally don't need a NEWS entry but when the CI check failed I wasn't sure if I read outdated info or the CI check is wrong :)

**Comment by taleinat** [comment]
Thanks for the PR, @zsol!

**Pull request merged**